src/commands/infra/generate.ts

```diff
@@ -60,61 +65,111 @@ export const generateCommandDecorator = (command: commander.Command): void => {
  * @param projectPath Path to the definition.yaml file
  */
 export const validateDefinition = async (
+  parentPath: string,
   projectPath: string
-): Promise<boolean> => {
+): Promise<string> => {
   try {
     // If templates folder does not exist, create cache templates directory
     mkdirp.sync(infraCommon.spkTemplatesPath);
-    if (!fs.existsSync(path.join(projectPath, "definition.yaml"))) {
-      logger.error(
-        `Provided project path for generate is invalid or definition.yaml cannot be found: ${projectPath}`
+    // Check for parent definition.yaml and child definition.yaml
+    if (
+      fs.existsSync(path.join(parentPath, "definition.yaml")) &&
+      fs.existsSync(path.join(projectPath, "definition.yaml"))
+    ) {
+      logger.warn(
+        `A definition.yaml was found in: ${parentPath}, and was found in ${projectPath}`
       );
-      return false;
+      return "A";
```

Comment:
nit: this can return something more descriptive like ParentLeafConfig instead of A and ParentConfig instead of B, just some name suggestions. It can be an enum.

Reply:
I think I am going to leave the configuration convention the way it is for now. Currently, the code docs explain what each configuration means, and this is actually only useful to the developer since it is not explicitly printed anywhere in the logs. When configurations become more complex, I will def keep this mind. Thanks!